internal/container/container.go

```diff
@@ -0,0 +1,249 @@
+package container
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+func newClient() (*client.Client, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Docker client: %w", err)
+	}
+
+	return cli, nil
+}
```

Comment:
It might be an idea to wrap this in `sync.Once` so we just open a single client. Something like:

```suggestion
var (
  dockerClientErr error
  dockerClient *client.Client
  dockerClientOnce sync.Once
)

func newClient() (*client.Client, error) {
  dockerClientOnce.Do(func() {
    dockerClient, dockerClientErr = client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
  })
  return dockerClient, dockerClientErr
}
```